### DISCUSSION TITLE
Lining up arrows with path ends

### DISCUSSION THREAD

**Discussion by LenAgain** [otfried/ipe#469] [General]
All my diagrams heavily use grid snapping, but the arrows on the end of polylines protrude past the end of the line and sometimes across other parts of the diagram.

I've tried changing the path join property but the rounded join has the same problem, and the bevel join is very ugly.

I also tried only filling the arrow and removing the stroke but the line that the arrow is on still shows underneath.
![image](https://github.com/otfried/ipe/assets/91378006/c69e65c8-7097-4a36-8d81-b3fffdcd345f)

Is there a simple way to have the tip of the arrow line up with the end of the path? This wouldn't really be an issue if my arrows were very large in proportion to the line width, but unfortunately that's not the case.

**Comment by otfried** [comment]
As you probably noticed by now, there are very few Ipe users watching this discussion forum.  You would reach them more easily using the Ipe discussion mailing list.  (Perhaps this will change in the future.)

I vaguely remember this issue coming up a long time in the past, but don't remember what we did about it.

I think there currently is no easy way to achieve what you want to do.   What I would do is:

1. On the line with the arrow, make the pen size "0.0"  (select the line, click on the button left of the pen settings, and simply enter 0 for the pen width).   This will make the arrow look as you want, but it will also make the line itself invisible.
2. Draw the line again (or duplicate the previous one).  Do not give this one an arrow.  This line will now stick out of the arrow like on your picture above.   So you need to edit this line to make it slightly shorter:  Select it, go into edit mode (Ctrl+E), right click on the right endpoint and select "Duplicate vertex".  This will create a new vertex that you can move inside the arrow head. Finally, delete the right endpoint (right-click and select "Delete vertex").

![image](https://github.com/otfried/ipe/assets/9983417/2f958711-efd4-48f9-ab2c-2ebbd02532dc)

Obviously one could make all this easier, but it seems it hasn't come up often.

**Comment by LenAgain** [comment]
Sorry to get back to this so late, I haven't been able to work on this project for a while.

The most elegant solution I've come up with is creating an arrow with a white box as the background and the actual arrow with no stroke. This way the arrow doesn't over-extend and the rest of the line that juts out the end is hidden by the box.
![image](https://github.com/otfried/ipe/assets/91378006/a4c653a2-0894-4e65-9136-1b431da46e63)
It doesn't preview perfectly in Ipe but the resulting PDF is convincing. Of course this then opens up various shenanigans with overlapping objects but this hasn't been to much of an issue so far.

I'm surprised more users haven't found this an issue given the grid-like nature of Ipe. It doesn't seem like it would be a straight forward fix though and I'm happy enough with the solution I've come to.
